CVAE_testbed/datasets/quilt_aics_features.py

```diff
@@ -0,0 +1,239 @@
+        try:
+            path_to_json = model_kwargs['json_quilt_path']
+        except:
+            path_to_json = "/home/ritvik.vasan/test/"
+        json_files = [pos_json for pos_json in os.listdir(path_to_json) if pos_json.endswith('.json')]
+
+        meta_to_file_name = []
+        for f in ds["cell_features"]:
+            meta_to_file_name.append({"filename": f, **ds["cell_features"][f].meta})
+        metas = pd.DataFrame(meta_to_file_name)
+
+        # Specify path to config file for FeatureDatabase
+        try:
+            db = FeatureDatabase(model_kwargs['config_path'])
+        except:
+            db = FeatureDatabase("/home/ritvik.vasan/config.json")
+
+        t = db.get_pg_table("featuresets", "aics-mitosis-classifier-four-stage_v1.0.0")
+
+        semi = metas.merge(t, left_on="CellId", right_on="CellId", suffixes=("_meta", "_mito"))
+
+        # Only interphase or no interphase
+        semi['Interphase and Mitotic Stages [stage]'] = semi['Interphase and Mitotic Stages [stage]'].apply(lambda x: 0 if x == 0.0 else 1)
+
+        dd = defaultdict(list)
+        for i in range(len(semi['filename'])):
+            this_file = semi['filename'][i]
+            a = json.loads(open('/home/ritvik.vasan/test/' + this_file).read())
+            a = dict([(key, value) for key,value in a.items() if key not in ['imsize_orig', 'com', 'angle', 'flipdim', 'imsize_registered' ] ])
+            a.update({'CellId': semi['CellId'][i]})
+            for key, value in a.items():
+                dd[key].append(value)
+
+        features_plus_cellid = pd.DataFrame(dict(dd))
+
+        meta_plus_features = pd.merge(semi, features_plus_cellid,on = 'CellId' )
+
+        i_care_cols = [c for c in meta_plus_features.columns if c not in ['CellId',
+                     'CellIndex', 'FOVId', 'WellId', 'FeatureExplorerURL',
+                     'CellLine', 'Workflow','associates','filename', 
+                     'NucMembSegmentationAlgorithm', 
+                     'NucMembSegmentationAlgorithmVersion', 
+                     'PlateId']]
+        
+        meta_plus_features = meta_plus_features[i_care_cols]
+        meta_plus_features.dropna(inplace=True)
+
+        categorical_features = ['Gene', 'ProteinDisplayName', 'StructureDisplayName']
+
+        categorical_dataframe = meta_plus_features[categorical_features]
+
+        non_categorical_dataframe = meta_plus_features[[c for c in meta_plus_features.columns if c not in categorical_features]]
+
+        one_hot_categorical_features = pd.get_dummies(categorical_dataframe, prefix = None, drop_first = True)
+        
+        num_of_cells = len(non_categorical_dataframe)
+        print(num_of_cells)
+
+        # Here we normalize X with min max
+        # x = non_categorical_dataframe.values
+        # min_max_scaler = preprocessing.MinMaxScaler()
+        # x_scaled = min_max_scaler.fit_transform(x)
+        # non_categorical_dataframe = pd.DataFrame(x_scaled)
```

Comment:
Why include this?

Reply:
trying different normalizations, need to clean this up.